e2e/ansible/playbooks/feature/ha/ha.yml

```diff
@@ -0,0 +1,206 @@
+           ctrl_node: "{{ result.stdout.split('/')[0].split('\t')[2]}}"
+
+       - name: Make the current ctrl node unschedulable  
+         shell: source ~/.profile; kubectl cordon {{ctrl_node}}
+         args:
+           executable: /bin/bash
+         register: result
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+      
+       - name: Confirm ctrl node is unschedulable 
+         shell: source ~/.profile; kubectl get nodes | grep {{ctrl_node}}
+         args:
+           executable: /bin/bash
+         register: result
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+         failed_when: "'SchedulingDisabled' not in result.stdout"
+
+       ########################################################
+       # INJECT FAULT                                         #
+       # (Includes using tools, commands etc.,)               #
+       ########################################################
+ 
+       - name: Delete the ctrl pod to force reschedule 
+         shell: source ~/.profile; kubectl delete pod {{ctrl_pod_name}}
+         args:
+           executable: /bin/bash
+         register: result
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+         failed_when: "ctrl_pod_name and 'deleted' not in result.stdout"
+
+       #########################################################
+       # VERIFY RECOVERY                                       #
+       # (Includes checking desired fault-tolerance behaviour) #
+       #########################################################
+ 
+       - name: Check if ctrl pod is restarted within 30s 
+         shell: >
+           source ~/.profile; 
+           kubectl get pods -l openebs/controller=jiva-controller
+         args:
+           executable: /bin/bash
+         register: result
+         until: "'Running' in result.stdout_lines[1]"
+         delay: 5
+         retries: 6
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+       #######################################################
+       # POST RECOVERY HEALTH-CHECK                          #
+       # (Includes application status etc.,)                 #
+       #######################################################   
+
+       - name: Get percona pod details post ha
+         shell: >
+           source ~/.profile; 
+           kubectl get pods -l name=percona -o wide
+           --no-headers
+         args:
+           executable: /bin/bash
+         register: result
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+       - name: Set pod variables to facts 
+         set_fact:
+           pod_status: "{{ result.stdout.split()[2] }}"
+           pod_restarts: "{{ result.stdout.split()[3] }}" 
+           pod_ip: "{{ result.stdout.split()[5] }}"
+
+       - name: Verify percona is running w/o restarts
+         debug: 
+           msg: "Percona pod health-check is successful"
+         failed_when: "pod_status != 'Running' and pod_restarts != '0'"
```

Comment:
Can we check through the history of messages or status to confirm ? There might be errors which were temporary but did not result in restarts ? Should we bother those cases ? If at all there were temporary issues can we capture for how long ?

Reply:
Can there be more verifications/checkpoints not limited to K8s ?

Reply:
Yes Amit, you are right. We **should** have non-k8s checkpoints to confirm successful HA. One sure-shot way - and the general expectation out of HA is : 

- To ensure no service breakage (no failure of mysql queries  in this case) during HA

For this, I am trying to identify the way to run parallel tasks via ansible - something I have alluded to in the comment about "async module" in the PR.  The plan is to have these added into the test once the method is identified. 

Now we are doing db writes after reschedule only as a post-ha recovery health-check.